ikarus/controlroutines/loadcontrol.hh

```diff
@@ -51,21 +51,41 @@ public:
           nonLinearSolver_->nonLinearOperator().lastParameter() = 0.0;
           nonLinearSolver_->nonLinearOperator().lastParameter() += 0.0;
         }, "The last parameter (load factor) must be assignable and incrementable with a double!");
+    if (loadSteps_ == 0)
+      DUNE_THROW(Dune::InvalidStateException, "Number of load steps should be greater than zero.");
   }
 
   /**
    * \brief Executes the LoadControl routine.
    *
    * \return ControlInformation structure containing information about the control results.
    */
-  ControlInformation run();
+  ControlState run();
 
 private:
   std::shared_ptr<NLS> nonLinearSolver_;
-  int loadSteps_;
+  size_t loadSteps_;
   double parameterBegin_;
   double parameterEnd_;
   double stepSize_;
+
+  /**
+   * \brief A wrapper function to update controlState after a nonlinear solver is executed successfully.
+   * The resulting controlState is then passed for further notifications.
+   *
+   * \param controlState Information about the state of the control routine.
+   * \param nonOp The non-linear operator.
+   * \param solverState State of the nonlinear solver.
+   */
+  void updateAndNotifyControlState(ControlState& controlState, typename NLS::NonLinearOperator& nonOp,
```

Comment:
This function just wraps certain repeated statements for better readability in `loadcontrol.inl`. The same is now implemented in `pathfollowing.inl`. I did not want to pass the control routine itself and have this function as a free function in some helper file, but perhaps the necessity of this as a free function in a separate file can be discussed.

Reply:
We keep this as such for now